typed-racket-lib/typed-racket/types/match-expanders.rkt

```diff
@@ -75,20 +75,19 @@
          (app (λ (t) (Listof? t #t)) (? Type? elem-pat)))])))
 
 
-(define-simple-macro (make-Listof-pred listof-pred?:id pair-matcher:id)
+(define-syntax-parse-rule (make-Listof-pred listof-pred?:id pair-matcher:id)
   (define (listof-pred? t [simple? #f])
     (match t
-      [(Mu-unsafe:
-        (Union: (== -Null)
-                (list (pair-matcher elem-t (B: 0)))))
+      [(Mu-unsafe: (Union: (== -Null) (list (pair-matcher elem-t (B: 0)))))
        (define elem-t* (instantiate-type elem-t t))
        (cond
          [simple? (and (equal? elem-t elem-t*) elem-t)]
          [else elem-t*])]
       [(Union: (== -Null) (list (pair-matcher hd-t tl-t)))
        (cond
          [(listof-pred? tl-t)
-          => (λ (lst-t) (and (equal? hd-t lst-t) hd-t))]
+          =>
+          (λ (lst-t) (and (equal? hd-t lst-t) hd-t))]
```

Comment:
The `define-simple-macro` migration rule shouldn't have bothered with these formatting changes. Filed jackfirth/resyntax#317.